addons/mrp/models/mrp_production.py

```diff
@@ -202,6 +203,21 @@ def _get_default_location_dest_id(self):
     production_location_id = fields.Many2one('stock.location', "Production Location", related='product_id.property_stock_production', readonly=False)
     picking_ids = fields.Many2many('stock.picking', compute='_compute_picking_ids', string='Picking associated to this manufacturing order')
     delivery_count = fields.Integer(string='Delivery Orders', compute='_compute_picking_ids')
+    cancel_need_confirmation = fields.Boolean(compute='_compute_cancel_need_confirmation')
+
+    @api.depends('move_raw_ids', 'move_finished_ids')
+    def _compute_cancel_need_confirmation(self):
+        domain = [
+            ('production_id', 'in', self.ids),
+            ('state', '=', 'done')
+        ]
+        res = self.env['stock.move'].read_group(domain, ['state', 'production_id'], ['production_id'])
+        ids = [rec['production_id'][0] for rec in res]
```

Comment:
doesn't change much but i prefer to index by dict
``` python
res = {rec['production_id'][0]: rec['state'] for rec in res}
for production in self:
     if recs.get(production.id):
        production.cancel_need_confirmation = True
    else:
        production.cancel_need_confirmation = False
```

Reply:
I did like that:
``` python
res = self.env['stock.move'].read_group(domain, ['state', 'production_id'], ['production_id'])
res = {rec['production_id'][0]: rec['production_id'] for rec in res}
for production in self:
    if res.get(production.id):
        production.confirm_cancel = True
    else:
        production.confirm_cancel = False
```